post-processor.go

```diff
@@ -268,14 +268,56 @@ func (p *PostProcessor) putManifest(manifest *Manifest) error {
 	if err := json.NewEncoder(&buf).Encode(manifest); err != nil {
 		return err
 	}
-	if err := p.s3.Put(p.config.ManifestPath, buf.Bytes(), "application/json", p.config.ACL); err != nil {
+
+	_, err := p.s3.PutObject(&s3.PutObjectInput{
+		Body:        strings.NewReader(buf.String()),
+		Bucket:      aws.String(p.config.Bucket),
+		Key:         aws.String(p.config.ManifestPath),
+		ContentType: aws.String("application/json"),
+		ACL:         aws.String(p.config.ACL),
+	})
+
+	if err != nil {
 		return err
 	}
+
 	return nil
 }
 
+func generateS3Url(region, bucket, key string) string {
+	return fmt.Sprintf("https://s3-%s.amazonaws.com/%s/%s", region, bucket, key)
+}
+
+// returns the size of the specified box, or -1 plus error on failure
+func getFileSize(boxPath string) (int64, error) {
+	// open the file
+	file, err := os.Open(boxPath)
+	if err != nil {
+		return -1, err
+	}
+
+	defer file.Close()
+
+	// return the file size
+	info, err := file.Stat()
```

Comment:
No need to open the file first, just use [os.Stat](https://golang.org/pkg/os/#Stat).